testing/test_config.py

```diff
@@ -1082,6 +1082,33 @@ def test_addopts_before_initini(self, monkeypatch):
         config._preparse([], addopts=True)
         assert config._override_ini == ["cache_dir=%s" % cache_dir]
 
+    def test_addopts_from_env_not_concatenated(self, monkeypatch):
+        """PYTEST_ADDOPTS should not take values from normal args (#4265)."""
+        from _pytest.config import get_config
+
+        monkeypatch.setenv("PYTEST_ADDOPTS", "-o")
+        config = get_config()
+        with pytest.raises(SystemExit) as excinfo:
+            config._preparse(["cache_dir=ignored"], addopts=True)
```

Comment:
Just curious, why do you need the `cache_dir=ignored`?

Reply:
Not needed, amended.

Reply:
Well, it is kind of needed, since previously it would be concatenated and then pass.

Reply:
Ahh I see, thanks!